### PR TITLE
 feat: support axis/legend `labelExpr` + add example with month initials

### LINKED ISSUE #5122: Support expression for formatting axis
We already have format types for number and time. We could introduce a flexible expression formatter. This would resolve https://github.com/vega/vega-lite-api/issues/12. 

Related issue: https://github.com/vega/vega/issues/608

cc @curran

### LINKED ISSUE #5249: Support "J F M A M J" as month labels

### PR DESCRIPTION
**Pull request by kanitw**
Fix #5122
Fix #5249

### CODE REVIEW

File: src/axis.ts [hunk 1]
```diff
@@ -185,6 +185,12 @@ export interface AxisOrientMixins {
 export type AxisConfig = VlOnlyGuideConfig & AxisOrientMixins & VgAxisConfigNoSignals;
 
 export interface Axis extends AxisOrientMixins, VgAxisConfigNoSignals, Guide {
+  /**
+   * Vega expression for customizing labels text.
+   * Note that the string label and value can be assessed via the `label` and `value` of the backing `datum` object.
```
Comment (domoritz) [inline src/axis.ts]:
```suggestion
   * Access the label text and value via the `label` and `value` of the `datum` object.
```

File: src/legend.ts [hunk 2]
```diff
@@ -110,6 +110,12 @@ export interface Legend
    */
   encoding?: LegendEncoding;
 
+  /**
+   * Vega expression for customizing labels text.
+   * Note that the string label and value can be assessed via the `label` and `value` of the backing `datum` object.
```
Comment (domoritz) [inline src/legend.ts]:
```suggestion
   * Access the label text and value via the `label` and `value` of the `datum` object.
```

File: test/compile/legend/assemble.test.ts [hunk 3]
```diff
@@ -1,6 +1,59 @@
 import {parseUnitModelWithScale} from '../../util';
 
 describe('legend/assemble', () => {
+  it('correctly applies labelExpr.', () => {
+    const model = parseUnitModelWithScale({
+      $schema: 'https://vega.github.io/schema/vega-lite/v3.json',
```
Comment (domoritz) [inline test/compile/legend/assemble.test.ts]:
```suggestion
```

File: test/compile/legend/assemble.test.ts [hunk 4]
```diff
@@ -1,6 +1,59 @@
...
+      description: 'A scatterplot showing horsepower and miles per gallons.',
```
Comment (domoritz) [inline test/compile/legend/assemble.test.ts]:
```suggestion
```

File: test/compile/legend/assemble.test.ts [hunk 5]
```diff
@@ -1,6 +1,59 @@
...
+
+    const legends = model.assembleLegends();
+    expect(legends[0].encode.labels.update.text).toEqual({
+      signal: 'datum.label[0]'
+    });
+  });
+
+  it('correctly applies labelExpr for timeUnit.', () => {
+    const model = parseUnitModelWithScale({
+      $schema: 'https://vega.github.io/schema/vega-lite/v3.json',
```
Comment (domoritz) [inline test/compile/legend/assemble.test.ts]:
```suggestion
```

File: test/compile/legend/assemble.test.ts [hunk 6]
```diff
@@ -1,6 +1,59 @@
...
+      description: 'A scatterplot showing horsepower and miles per gallons.',
```
Comment (domoritz) [inline test/compile/legend/assemble.test.ts]:
```suggestion
```

**Comment by domoritz** [comment]
Why didn't we add this for header?

**Comment by kanitw** [comment]
Headers require different code and I can't think of a convincing use case for headers `labelExpr` to justify the implementation effort.

**Pull request merged**